impl/maven-cli/src/main/java/org/apache/maven/cling/invoker/mvnenc/EncryptInvoker.java

```diff
@@ -78,8 +78,8 @@ protected int execute(EncryptContext context) throws Exception {
             context.addInHeader("This tool is part of Apache Maven 4 distribution.");
             context.addInHeader("");
 
-            Thread executeThread = Thread.currentThread();
-            context.terminal.handle(Terminal.Signal.INT, signal -> executeThread.interrupt());
+            context.terminal.handle(
+                    Terminal.Signal.INT, signal -> Thread.currentThread().interrupt());
```

Comment:
That sounds weird.  The current thread cannot be the Maven execution thread, unless maybe it actually reads from stdin.  It should be a terminal's pump thread.

Reply:
mvnenc command is always single threaded, and did install this handler from beginning to handle ctrl+c while inside of ConsolePrompt (ESC is back, Ctrl+C is "abort").